src/main/java/dev/ebullient/convert/tools/JsonNodeReader.java

```diff
@@ -27,21 +29,21 @@ static <E extends Enum<E>> E getEnumValue(String value, Class<E> enumClass) {
         if (!isPresent(value)) {
             return null;
         }
-        if (FieldValue.class.isAssignableFrom(enumClass)) {
-            // If it's a FieldValue, then try to use the matches() method instead.
-            return Arrays.stream(enumClass.getEnumConstants())
-                    .filter(e -> ((FieldValue) e).matches(value.toLowerCase()) || ((FieldValue) e).matches(value.toUpperCase()))
-                    .findAny().orElse(null);
-        }
         try {
             return Enum.valueOf(enumClass, value.toLowerCase());
         } catch (IllegalArgumentException ignored) {
         }
         try {
             return Enum.valueOf(enumClass, value.toUpperCase());
         } catch (IllegalArgumentException ignored) {
-            return null;
         }
+        // Try case-insensitive matching, unless we have a matches method to use
+        Predicate<E> filterFunc = FieldValue.class.isAssignableFrom(enumClass)
+            ? e -> ((FieldValue) e).matches(value)
+            : e -> e.name().equalsIgnoreCase(value);
+        return Arrays.stream(enumClass.getEnumConstants())
+            .filter(filterFunc)
+            .findAny().orElse(null);
     }
```

Comment:
Was the order/precedence change here intentional?

Reply:
It was intentional to prioritize the `FieldValue.matches` implementation over the default behavior. My thinking is that if it's been specifically overridden in the matches method, we might not want the default behavior at all.